src/Raven.Studio/typescript/components/common/select/Select.tsx

```diff
@@ -46,14 +52,15 @@ export default function Select<
     return <ReactSelect {...props} className="bs5 react-select-container" classNamePrefix="react-select" />;
 }
 
-export function OptionWithIcon(props: OptionProps<SelectOptionWithIcon>) {
-    const { data } = props;
+export function OptionWithIcon(props: OptionProps<SelectOptionWithIcon<any>>) {
```

Comment:
This component was only responsible for rendering options with an icon. Let's restore the SelectOptionWithIcon type.